lib/autoload/kuroko2/workflow/notifier/webhook.rb

```diff
@@ -0,0 +1,172 @@
+module Kuroko2
+  module Workflow
+    module Notifier
+      class Webhook
+        attr_reader :message_builder
+
+        HASH_ALGORITHM = 'sha256'
+        HMAC_DIGEST    = OpenSSL::Digest.new(HASH_ALGORITHM)
+
+        def initialize(instance)
+          @instance   = instance
+          @definition = instance.job_definition
+          @message_builder = Workflow::Notifier::Concerns::ChatMessageBuilder.new(instance)
+          @secret_token = Kuroko2.config.notifiers.webhook.try!(:secret_token)
+        end
+
+        def notify_launch
+          if @definition.hipchat_notify_finished?
+            request(
+              build_payload(
+                action: 'notify_launch',
+                level: 'INFO',
+                subject: message_builder.launched_text,
+                message: @instance.logs.select{ |log| log.level == 'INFO' }.last.try!(:message),
+              )
+            )
+          end
+        end
+
+        def notify_retrying
+          if @definition.hipchat_notify_finished?
+            request(
+              build_payload(
+                action: 'notify_retrying',
+                level: 'INFO',
+                subject: message_builder.retrying_text,
+                message: @instance.logs.last(2).first.message,
```

Comment:
The `logs` is a `ActiveRecord::Associations::CollectionProxy`.
IF `logs` size is 1, `logs.last(2).first` returns results. But `logs[-2]` returns `nil`.
I want to keep this `logs.last(2).first` behavior.